### PR TITLE
create channels list and track page :rocket:

### PR DESCRIPTION
**Pull request by Tanzeel-khan**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "dependencies": {
     "@emotion/react": "^11.10.4",
     "@emotion/styled": "^11.10.4",
+    "@mui/icons-material": "^5.10.9",
     "@mui/material": "^5.10.8",
     "@testing-library/jest-dom": "^5.16.5",
```
Comment (ecspress) [inline package.json]: dev dependencies (testing, types, ts) should not be listed in dependencies

File: package.json [hunk 2]
```diff
@@ -5,6 +5,7 @@
   "dependencies": {
     "@emotion/react": "^11.10.4",
     "@emotion/styled": "^11.10.4",
+    "@mui/icons-material": "^5.10.9",
     "@mui/material": "^5.10.8",
```
Comment (ecspress) [inline package.json]: why do we need emotion/* and styledcomponents both?

Comment (ecspress) [inline package.json:7]: if we are going to use 3rd party ui library, then all components need to be wrapped.

File: src/App.tsx [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
+import Tracks from './container/tracks-container';
+import Home from './screens/Home';
 
 
 function App() {
   return (
     <Routes>
-        <Route path = "/" element={<Tracks/>}/>
+        <Route path = "/" element={<Home/>}/>
+        <Route path = "/channels/:channelName" element={<Tracks/>}/>
```
Comment (ecspress) [inline src/App.tsx]: routes need to be in their own file.

File: src/components/Channels/index.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
+import React from 'react'
+import { useNavigate } from 'react-router-dom';
+import styled from 'styled-components';
+import { Channels } from '../../types/channel';
+import { mobile } from '../../utils/responsive';
```
Comment (ecspress) [inline src/components/Channels/index.tsx]: setup project to avoid using relative imports '../../../'

File: src/components/Channels/index.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  background-color: rgba(0, 0, 0, 0.767);
+  z-index: 3;
```
Comment (ecspress) [inline src/components/Channels/index.tsx]: fix formatter in ide to autoformat

File: src/components/Channels/index.tsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+  width: 100%;
+  height: 100%;
+  object-fit: fill;
+  cursor: pointer;
+  opacity: 0.7;
+  ${mobile({ height: "20vh" })};
+`;
+
+const Title = styled.h1`
+    color:white;
```
Comment (ecspress) [inline src/components/Channels/index.tsx]: move theme related info to theme file (colors, font, ...)

File: src/components/Channels/index.tsx [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+transition: all 0.5s ease;
+&:hover {
+  background-color: #e9f5f5;
+  transform: scale(1.1);
+}
+`;
+const navigate = useNavigate();
+const navigateChannel = () => {
+  navigate(`${channelRoute}`);
+}
```
Comment (ecspress) [inline src/components/Channels/index.tsx]: dont navigate from component

File: src/components/Channels/index.tsx [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+  return (
+    <>
+    <Container>
+        <Image src={img} />
+        <Info onClick={navigateChannel}>
+          <Title>{title}</Title>
+        </Info>
+    </Container>
+    </>
```
Comment (ecspress) [inline src/components/Channels/index.tsx]: dont use fragment for only one child

**Review by ecspress — CHANGES_REQUESTED** [review]

File: .firebase/hosting.YnVpbGQ.cache [hunk 9]
```diff
@@ -0,0 +1,24 @@
+asset-manifest.json,1665773265084,c23a589488f08c6caa05c63a6c216cfc494831f60a4cde9d81c93c5120e8db54
```
Comment (aitchkhan) [inline .firebase/hosting.YnVpbGQ.cache:1]: add .firebase to .gitignore and remove it from commit

File: src/components/Footer/index.tsx [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+const Center = styled.div`
+  flex: 1;
+  text-align: center;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  padding: 20px;
+  ${mobile({ padding: "10px", marginTop: "-15px" })};
+`;
```
Comment (ecspress) [inline src/components/Footer/index.tsx:26]: merge these styles

File: src/components/Hero/index.tsx [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+    })};
+  `;
+
+  return (
+    <>
+      <Heading>Podcast Network</Heading>
+      <ShortDescription>
+        An Interactive Streaming Platform For Muslim Ummah
+      </ShortDescription>
+    </>
```
Comment (ecspress) [inline src/components/Hero/index.tsx:35]: use a string file for text strings

File: src/types/tracks.ts [hunk 12]
```diff
@@ -0,0 +1,7 @@
+export interface AudioTracks {
+    audioFunction: (e:any) => void;
+    channelTracks: any;
```
Comment (ecspress) [inline src/types/tracks.ts:3]: dont use any

File: src/container/tracks-container.tsx [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+  channelTracks,
+  scholarDesc,
+  scholarName,
+  channelImg,
+}: AudioTracks) => {
+  return (
+    <>
+      <Header />
+      <Container>
+        <Card sx={{ display: "flex", maxWidth: "800px" }}>
```
Comment (ecspress) [inline src/container/tracks-container.tsx:62]: no inline styles if we are using styled-components

File: src/routes.tsx [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+  const handleClickNext = () => {
+    setTrackIndex((currentTrack: any) =>
+      currentTrack < channelTracks.length - 1 ? currentTrack + 1 : 0
+    );
+  };
+  const Tracksrc = channelTracks[trackIndex]?.src;
+  const Trackname = channelTracks[trackIndex]?.name;
+  const scholarName = channelData[channelIndex]?.scholarName;
+  const scholarDesc = channelData[channelIndex]?.scholarDescription;
+  const channelImg = channelData[channelIndex]?.img;
```
Comment (ecspress) [inline src/routes.tsx:57]: can we use destructuring here?

**Review by ecspress — COMMENTED** [review]
Add prettier formatter to format code.

**Pull request merged**